src/utils/network.h

```diff
@@ -137,11 +137,31 @@ static inline int pg_utils_get_ipv6_l4(struct rte_mbuf *pkt, uint8_t *ip_type,
 			loop = false;
 		}
 	}
-	*ip_type = next_header;
-	*ip_payload = next_data;
+	if (!size) {
+		*ip_type = next_header;
+		*ip_payload = next_data;
+	} else {
+		*size += next_data - tmp;
+	}
 	return 0;
 }
 
+static inline int pg_utils_get_l3_len(struct rte_mbuf *pkt)
```

Comment:
same thing here: this function work only with ipv6, so please keep "ipv6" in the function name